From 30c8bd253a8dae0946dc0be03b4eb13dbd273122 Mon Sep 17 00:00:00 2001
From: Dreamail <dreaillll@outlook.com>
Date: Wed, 16 Apr 2025 22:10:48 +0800
Subject: [PATCH] remove session type check

since some wm run in type greeter and has a manager type but it will cause a bug
---
 src/actions.rs | 33 ++++++++++++---------------------
 1 file changed, 12 insertions(+), 21 deletions(-)

diff --git a/src/actions.rs b/src/actions.rs
index fc7a789..09d7a61 100644
--- a/src/actions.rs
+++ b/src/actions.rs
@@ -633,32 +633,23 @@ async fn graphical_user_sessions_exist(
             .await
             .map_err(|e| warn!("graphical_user_sessions_exist: builder: {e:?}"))
         {
-            if let Ok(class) = session_proxy.class().await.map_err(|e| {
-                warn!("graphical_user_sessions_exist: class: {e:?}");
+            if let Ok(type_) = session_proxy.type_().await.map_err(|e| {
+                warn!("graphical_user_sessions_exist: type_: {e:?}");
                 e
             }) {
-                if class == SessionClass::User {
-                    if let Ok(type_) = session_proxy.type_().await.map_err(|e| {
-                        warn!("graphical_user_sessions_exist: type_: {e:?}");
-                        e
-                    }) {
-                        match type_ {
-                            SessionType::X11 | SessionType::Wayland | SessionType::MIR => {
-                                if let Ok(state) = session_proxy.state().await.map_err(|e| {
-                                    warn!("graphical_user_sessions_exist: state: {e:?}");
-                                    e
-                                }) {
-                                    match state {
-                                        SessionState::Online | SessionState::Active => {
-                                            return Ok(true)
-                                        }
-                                        SessionState::Closing => {}
-                                    }
-                                }
+                match type_ {
+                    SessionType::X11 | SessionType::Wayland | SessionType::MIR => {
+                        if let Ok(state) = session_proxy.state().await.map_err(|e| {
+                            warn!("graphical_user_sessions_exist: state: {e:?}");
+                            e
+                        }) {
+                            match state {
+                                SessionState::Online | SessionState::Active => return Ok(true),
+                                SessionState::Closing => {}
                             }
-                            _ => {}
                         }
                     }
+                    _ => {}
                 }
             }
         }
-- 
2.48.1

